### PR TITLE
Generate `torch._C._nn` functions from yaml

### LINKED ISSUE #103787: Generate complete annotations for `torch._C._nn`
### 🚀 The feature, motivation and pitch

This is a follow-up of #102918 and a sub-task of #103761.

This is not user-facing and only involves changes in the type stub of `torch._C._nn` which is an internal module.

#102918 has introduced annotations for some functions in `torch._C._nn` so that the end-users can see correct typing for public APIs exposed through `torch.nn.functional`. There's also a TODO left in that PR because there're many functions not covered by it which do not directly contribute to the publicly visible APIs.
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/tools/pyi/gen_pyi.py#L299

However, this is not enough for the goal in #103761. When we remove the pyi type stub, mypy will look into more details of `torch.nn.functional` and throw errors about undefined attributes in `torch._C._nn`, which are only internally used and not covered by #102918.

### Alternatives

Add a hundred `type:ignore`s in `torch/nn/functional.py` when working on #103761. (Obviously not good)

### Additional context

There should be a better way to engineer it -- instead of manually annotating each function, it could be possible to generate the function signatures automatically. I'm currently looking into it.

### PR DESCRIPTION
**Pull request by lkct**
Fixes #103787

- Manually appended annotations in `gen_pyi.py` for `torch._C._nn` are now fully generated.
- Some functions manually added in `torch/_C/_nn.pyi.in` are now generated.
- Runtime check `'mkldnn_prelu' in dir(torch._C._nn)` is `False`, therefore it's removed.

### CODE REVIEW

**Comment by lkct** [comment]
`torch._C._nn` is indeed not user-facing.

@pytorchbot label "topic: not user facing" "topic: improvements"

**Comment by janeyx99** [comment]
@bdhirsh would you be the right reviewer here?
